corehq/apps/api/resources/v0_5.py

```diff
@@ -1089,7 +1089,15 @@ def to_obj(action_times):
 
     def alter_list_data_to_serialize(self, request, data):
         data['meta']['local_date_timezone'] = self.local_timezone.zone
+        params = request.GET.copy()  # Makes params mutable for creating next_url below
 
+        if data['meta']['total_count'] >= data['meta']['limit']:
```

Comment:
Previously, the limiting was done directly on the query. So total_count would at max be equal to the limit. 
But I changed it to just use the built in tastypie limit to constrain the number of objects return so corrected that conditional to be `total_count > limit`
https://github.com/dimagi/commcare-hq/pull/32922/commits/dfbc42ad74155ec302d6784f96a03ebfc9e46824

Reply:
I see you switched this back to `>=` when reverting the `limit` change.  Can you walk me through this again?  Does `total_count` just return the length of the list returned from `obj_get_list`?  If so, should it instead be returning the results without the limit applied?

What if `obj_get_list` returns a queryset instead of a list, would that let this work as desired?  Ie, get the correct total count but not fetch all results directly.  Alternatively, is there a `total_count` function or something to override?

(It may also be the case that we don't care about `total_count` at all, which is probably acceptable too)

Reply:
D&A would want the total_count. We're able to directly override the value within `alter_list_data_to_serialize`
[be0fed8](https://github.com/dimagi/commcare-hq/pull/32922/commits/be0fed841a55eb6fb0b8ddff4c1054ab76c5eb55)

Reply:
> the queryset needs to be evaluated within the with override_settings(USE_TZ=True): block

Ah right, I'd forgotten about that, that makes sense then.  I think the "tastypie" approach would be to override the paginator class with something custom, but that's probably way too much boilerplate for a one-off thing like this (though I guess the cursor stuff could go in there too).  This is one of the things I find frustrating with tastypie - it's nice and elegant up front, but then you end up needing to understand the whole complicated system and override a bunch of stuff to do anything non-standard anyways.

Anyways, I agree with the approach you took here.